lib/improver/tests/set_up_test_cubes.py

```diff
@@ -211,6 +211,13 @@ def set_up_variable_cube(data, name='air_temperature', units='K',
                 raise ValueError(
                     'Cannot generate {} realizations from data of shape '
                     '{}'.format(len(realizations), data.shape))
+            realizations = np.array(realizations)
+            if issubclass(realizations.dtype.type, np.integer):
+                # expect integer realizations
+                realizations = realizations.astype(np.int32)
+            else:
```

Comment:
Why do you need this exception? Can't you just reset the datatypes when you convert this coordinate to a percentile or threshold coordinate? I think you might already be doing this anyway.

Reply:
We need this exception because when this function is called as part of `set_up_percentile_cube` or `set_up_probability_cube`, the values of this coordinate (which is renamed to `percentile` or `threshold`) may not actually be integers.  Eg you might want thresholds of 0.5C, 1C, 1.5C, etc.  So casting them to integers at this point changes the values.

This is not the most robust code design in general, but since it's only for unit test setup I thought it was acceptable to do it this way.

Reply:
When I was trying to understand this I noticed you are also enforcing the datatype of the percentile or threshold coordinate in the set_up_percentile_cube and set_up_probability_cube functions. This is no longer needed with this change.  You could change these functions to put the data into the coordinate with the right data types, rather then putting the fix in here. But up to you.

Reply:
OK.